[docs] Add unstyled section to all components coming with the package (#25843)
mui-org/material-ui: docs/src/pages/components/portal/portal.md

React [doesn't support](https://github.com/facebook/react/issues/13097) the [`createPortal()`](https://reactjs.org/docs/portals.html) API on the server.
You have to wait for the client-side hydration to see the children.

## Unstyled

> As a component that does not have any styles, the portal also comes with the unstyled package.

Reviewer: I like this. What do you think about using this exact wording for https://next--material-ui.netlify.app/components/trap-focus/#unstyled?

```suggestion
As the component that does not have any styles, it also comes with the unstyled package.
```

Author: Maybe
```suggestion
As the component does not have any styles, the portal also comes with the unstyled package.
```

?

Reviewer: It looks the same :)
